reconcile/utils/terrascript_aws_client.py

```diff
@@ -1258,10 +1254,36 @@ def populate_tf_resource_rds(self, spec):
             pg_values["parameter"] = pg_values.pop("parameters")
             if self._multiregion_account(account) and len(provider) > 0:
                 pg_values["provider"] = provider
-            pg_tf_resource = aws_db_parameter_group(pg_identifier, **pg_values)
+            return aws_db_parameter_group(pg_identifier, **pg_values)
+
+        # 'deps' should contain a list of terraform resource names
+        # (not full objects) that must be created
+        # before the actual RDS instance should be created
+        deps = []
+
+        parameter_group = values.pop("parameter_group", None)
+        if parameter_group:
+            pg_tf_resource = populate_parameter_group(parameter_group)
             tf_resources.append(pg_tf_resource)
-            deps = self.get_dependencies([pg_tf_resource])
-            values["parameter_group_name"] = pg_name
+            deps += self.get_dependencies([pg_tf_resource])
+            # Associate parameter group to db instance
+            values["parameter_group_name"] = pg_tf_resource.get("name")
+
+        old_parameter_group = values.pop("old_parameter_group", None)
+        if old_parameter_group:
+            # discourage using old_parameter_group if parameter_group field is not utilized.
+            if parameter_group is None:
+                raise ValueError(
+                    "Cannot use old_parameter_group field without parameter_group."
+                    "This field is only used during RDS major version upgrade"
+                )
+
+            old_pg_tf_resource = populate_parameter_group(old_parameter_group)
+
+            if old_pg_tf_resource.get("name") == pg_tf_resource.get("name"):
+                raise ValueError("Must supply a unique name value for parameter group")
```

Comment:
The only other thing that I would add here is that it might save us time to make this error message a little bit more descriptive, particularly about how to fix it.